client/src/js/directives/findpatient.js

```diff
@@ -1,5 +1,20 @@
 angular.module('bhima.directives')
-.directive('findPatient', ['$compile', '$http', 'validate', 'messenger', 'appcache', function($compile, $http, validate, messenger, AppCache) {
+.directive('findPatient', FindPatientDirective);
+
+FindPatientDirective.$inject = [
+'$compile', '$http', 'validate', 'messenger', 'appcache'
+];
+
+/**
+* The FindPatient directive allows a user to serach for a patient by either the
+* patient identifier (Project Abbreviation + Reference) or by typeahead on patient
+* name.
+*
+* The typeahead loads data as your type into the input box, pinging th URL
+* /patient/search/fuzzy (for legacy reasons, not actually a fuzzy search).  The
+* HTTP endpoints sends back 10 results which are then presented to the user.
+*/
+function FindPatientDirective($compile, $http, validate, messenger, AppCache) {
   return {
     restrict: 'AE',
     templateUrl : 'partials/templates/findpatient.tmpl.html',
```

Comment:
```
function searchReference(ref) {
        var url = '/patient/search/reference/';
        $http.get(url + ref)
        .success(selectPatient)
        .error(function (err) {
          console.error(err); 
        })
        .finally();
      }
```
As discussed, the **then, catch** is more recommanded because .success will be deprecated.